src/js/stores/MesosStore.js

```diff
@@ -0,0 +1,247 @@
+var EventEmitter = require("events").EventEmitter;
+var semver = require("semver");
+
+var AppDispatcher = require("../AppDispatcher");
+var Util = require("../helpers/Util");
+var InfoStore = require("../stores/InfoStore");
+var InfoActions = require("../actions/InfoActions");
+var InfoEvents = require("../events/InfoEvents");
+var MesosActions = require("../actions/MesosActions");
+var MesosEvents = require("../events/MesosEvents");
+
+const FILES_TTL = 60000;
+const STATE_TTL = 60000;
+const REQUEST_DELAY = 100;
+const MASTER_ID = "master";
+
+var version = null;
+var stateMap = {};
+var taskFileMap = {};
+var taskFileRequestQueue = [];
+var requestCount = 0;
+
+function getDataFromMap(id, map, ttl = 100) {
+  if (!Util.isString(id) || map == null) {
+    return null;
+  }
+
+  let item = map[id];
+  if (item == null) {
+    return null;
+  }
+  if (item.timestamp + ttl < Date.now()) {
+    invalidateMapData(id, map);
+    return null;
+  }
+
+  return item.data;
+}
+
+function invalidateMapData(id, map) {
+  if (!Util.isString(id) || map == null) {
+    return;
+  }
+  delete map[id];
+}
+
+function addDataToMap(id, map, data) {
+  if (!Util.isString(id) || map == null || data == null) {
+    return;
+  }
+  map[id] = {
+    data: data,
+    timestamp: Date.now()
+  };
+}
+
+var MesosStore = Object.assign({
+
+  getState: function (nodeId) {
+    return getDataFromMap(nodeId, stateMap, STATE_TTL);
+  },
+
+  getTaskFiles: function (taskId) {
+    return getDataFromMap(taskId, taskFileMap, FILES_TTL);
+  }
+
+}, EventEmitter.prototype);
+
+function getNodeURLFromState(nodeId, state) {
+
+  if (state == null) {
+    return null;
+  }
+
+  let agent = state.slaves.find((slave) => {
+    return slave.id === nodeId;
+  });
+
+  if (agent == null) {
+    return null;
+  }
+
+  let pid = agent.pid;
+  let hostname = agent.hostname;
+  return `//${hostname}:${pid.substring(pid.lastIndexOf(":") + 1)}`;
+}
+
+function getExecutorDirectoryFromState(frameworkId, taskId, state) {
+
+  if (state == null) {
+    return null;
+  }
+
+  function matchFramework(framework) {
+    return frameworkId === framework.id;
+  }
+
+  let framework = state.frameworks.find(matchFramework) ||
+    state.completed_frameworks.find(matchFramework);
+
+  if (framework == null) {
+    return null;
+  }
+
+  function matchExecutor(executor) {
+    return taskId === executor.id;
+  }
+
+  let executor = framework.executors.find(matchExecutor) ||
+    framework.completed_executors.find(matchExecutor);
+
+  if (executor == null) {
+    return null;
+  }
+
+  return executor.directory;
+}
+
+function throttleRequest(request) {
```

Comment:
+1 on that. Let's go for the best-effort approach, everything else seems a bit **overkill** for the moment. Ideally though, we should somehow inform the user that the logs are not available at the time, and they should be able to retry later. Maybe we can offer a "retry" link?